src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -1873,4 +1881,101 @@ private boolean expectedFunction(@Nullable PsiElement resolve) {
       return true;
     }
   }
+
+  public static class ErlangFunctionCallNamedParameter<T extends PsiElement> extends PatternCondition<T> {
+    private final List<String> myNames;
+
+    public ErlangFunctionCallNamedParameter(@NotNull String... names) {
+      super("functionCallNamedParameter");
+      myNames = ContainerUtil.newArrayList(names);
+    }
+
+    @Override
+    public boolean accepts(@NotNull T element, ProcessingContext context) {
+      ErlangExpression expr = PsiTreeUtil.getParentOfType(element, ErlangExpression.class);
+      ErlangArgumentList args = ObjectUtils.tryCast(expr != null ? expr.getParent() : null, ErlangArgumentList.class);
+      if (expr == null || args == null) return false;
+
+      int position = args.getExpressionList().indexOf(expr);
+      if (position == -1) return false;
+
+      ErlangFunctionCallExpression call = ObjectUtils.tryCast(args.getParent(), ErlangFunctionCallExpression.class);
+      ErlangFunction function = resolveToFunction(call);
+      if (function == null) return false;
```

Comment:
The code which obtains a function call given call's parameter is now duplicated three times: in your two patterns, and in ```org.intellij.erlang.psi.impl.ErlangPsiImplUtil.ErlangFunctionCallParameter``` as well. Do something about it.